packages/mdc-menu-surface/foundation.ts

```diff
@@ -0,0 +1,500 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+// tslint:disable:no-bitwise object-literal-sort-keys
+
+import MDCFoundation from '@material/base/foundation';
+import {MDCMenuSurfaceAdapter} from './adapter';
+import {Corner, CornerBit, cssClasses, MenuDimensions, MenuDistance, MenuPoint, numbers, strings} from './constants';
+
+interface AutoLayoutMeasurements {
+  viewportSize: MenuDimensions;
+  viewportDistance: MenuDistance;
+  anchorSize: MenuDimensions;
+  surfaceSize: MenuDimensions;
+  bodySize: MenuDimensions;
+  windowScroll: MenuPoint;
+}
+
+class MDCMenuSurfaceFoundation extends MDCFoundation<MDCMenuSurfaceAdapter> {
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  static get strings() {
+    return strings;
+  }
+
+  static get numbers() {
+    return numbers;
+  }
+
+  static get Corner() {
+    return Corner;
+  }
+
+  /**
+   * {@see MDCMenuSurfaceAdapter} for typing information on parameters and return types.
```

Comment:
is this a closure comment?

Reply:
This comment already existed in `foundation.js`. Updated it to use proper [JSDoc syntax](http://usejsdoc.org/tags-see.html).